### ISSUE TITLE
gRPC server streaming: stream terminated by RST_STREAM with error code: NO_ERROR

### ISSUE THREAD

**Issue by DanTulovsky** [datawire/ambassador#3154]
**Describe the bug**
I have the following setup:

grpc client (local workstation) -> loadbalancer (digital ocean, TCP) -> ambassador (kubernetes cluster) -> grpc backend

Regular (non-streaming) grpc calls work just fine.  If I try to make a streaming call, the first time the server tries to send data back on it, I get the following error:

```
rpc error: code = Internal desc = stream terminated by RST_STREAM with error code: NO_ERROR
```

If I remove ambassador from the chain and send traffic directly from the loadbalancer to the backend, the server streaming rpc works just fine.

**Expected behavior**
Streaming RPC works properly.

**Versions (please complete the following information):**
 - Ambassador: Edge stack 1.10.0
 - Kubernetes environment: Digital Ocean 
 - Version:  1.19.3-do.3

**Additional context**

Ambassador host config:

```
apiVersion: getambassador.io/v2
kind: Host
metadata:
  name: pepper-poker-grpc.wetsnow.com
  namespace: ambassador
spec:
  acmeProvider:
    email: dant@wetsnow.com
  requestPolicy:
    insecure:
      action: Reject
  hostname: pepper-poker-grpc.wetsnow.com
  tlsSecret:
    name: pepper-poker-grpc.wetsnow.com
---
apiVersion: getambassador.io/v2
kind: TLSContext
metadata:
  name: pepper-poker-grpc.wetsnow.com-context # for the host above
  namespace: ambassador
spec:
  hosts:
    - "pepper-poker-grpc.wetsnow.com"
  secret: pepper-poker-grpc.wetsnow.com
  alpn_protocols: h2,http/1.1 # required for C# grpc client
```

Ambassador mapping config:

```
apiVersion: getambassador.io/v2
kind: Mapping
metadata:
  name: pepper-poker-grpc-server-mapping
  namespace: ambassador
spec:
  prefix: /poker.PokerServer/
  rewrite: /poker.PokerServer/
  service: https://pepper-poker-grpc-service.poker:8443
  timeout_ms: 0 # required for streaming grpc
  grpc: True
  headers:
    :authority: pepper-poker-grpc.wetsnow.com
  resolver: endpoint
  load_balancer:
    policy: least_request
```

Ambassador global config:

```
apiVersion: getambassador.io/v2
kind: Module
metadata:
  name: ambassador
  namespace: ambassador
spec:
  config:
    use_proxy_proto: true
    envoy_log_type: json
    add_linkerd_headers: true
```

Note that I've tried turning off proxy proto (both in ambassador and loadbalanacer) and it didn't matter.  I tried to disable linkerd, and that didn't matter either.

Oddly enough though, this used to work a few weeks ago, unfortunately I am not sure what exactly changed to break it, since I didn't catch the break in time (this is just a toy project, so no proper monitoring or any real traffic).

No filters or policies for this host are enabled.

(Side question, when integrating with linkerd, should the linkerd sidecars be injected for the amabassador pods themselves?)

Thank you
Dan

**Comment by DanTulovsky** [comment]
Ok, so it looks like the problem here is the `timeout_ms' setting.  

```
timeout_ms: 0
```

is supposed to disable the timeout, but it does not. Instead the timeout is the default 3000ms.  I see the connection dropping exactly 3 seconds after starting. I can set it to some other values and it properly waits that long.

Looks like this is a regression?

**Comment by cakuros** [comment]
Hi, @DanTulovsky

can you try setting `AMBASSADOR_LEGACY_MODE: true` in the deployment env variables and see if that resolves the issue?

**Comment by DanTulovsky** [comment]
Yes, that does fix it.

**Comment by khussey** [comment]
This was fixed in Ambassador 1.11.2, which is now available.
